src/chapters/orm-relationships/one-to-many.rst

```diff
@@ -0,0 +1,198 @@
+
+On lines 37-38 we use this fact to check to see if an ``EventCategory`` object was found. If not, we display an error message in the title.
+
+.. admonition:: Tip
+
+   As you wade into vast waters of Java and Spring, you will frequently find yourself needing to use a method or class that you have not encountered before. In such cases, be brave and remember that you don't have understand *all* of the documentation for a method or class in order to use it. 
+
+Update 4: Retrieving Events For the Given Category
+++++++++++++++++++++++++++++++++++++++++++++++++++
+
+We are finally ready to consider the full conditional:
+
+.. sourcecode:: java
+   :lineno-start: 32
+
+   if (categoryId == null) {
+      model.addAttribute("title", "All Events");
+      model.addAttribute("events", eventRepository.findAll());
+   } else {
+      Optional<EventCategory> result = eventCategoryRepository.findById(categoryId);
+      if (result.isEmpty()) {
+            model.addAttribute("title", "Invalid Category ID: " + categoryId);
+      } else {
+            EventCategory category = result.get();
+            model.addAttribute("title", "Events in category: " + category.getName());
+            model.addAttribute("events", category.getEvents());
+      }
+   }
+
+Lines 39-42 will only execute if we know for sure that a category with the given ID exists. In this case, we extract the category from the ``Optional`` object ``result`` using the ``get()`` method. Then we create an appropriate title using the category name, and pass the events belonging to the category into the view.
+
+Line 42 is what we have been building up to this entire time. Our ``category`` object now has an ``events`` field that is populated using the ``@OneToMany`` annotation that we added near the beginning of this section, allowing the inverse relationship to be established.
+
+Now, not only do events have categories, but categories know which events they contain.
+
+Linking to Filtered Views
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+With our new controller logic in place, we have one small update to make before we can fully use this new functionality.
+
+We need to provide a way for the user to access our new filtered views. In the ``eventCategories/index.html`` template, modify the loop within the table to include a link to the filtered view for each category:
+
+.. sourcecode:: html
+   :lineno-start: 14
+
+   <tr th:each="category : ${categories}">
+      <td><a th:text="${category.name}" th:href="'/events?categoryId=' + ${category.id}"></a></td>
+   </tr>
+
+The logic in ``th:href`` uses string concatenation to create the appropriate URL for each category.
+
+With this code in place, we can start up our application and test out the new functionality! Be sure to navigate to ``/eventCategories`` to see your new links to the filtered views. And also be sure to test that ``/events`` still displays all events. 
```

Comment:
"With this code in place, start up the application and test..."

Remove the "And" in the last sentence. It sounds better as, "Also, be sure to test..."